spec/spec_helper.rb

```diff
@@ -114,11 +115,23 @@ class Pet < ActiveRecord::Base
   col :home_address, :type => :text
   if ENV['DB'] == 'postgresql'
     col :tsntz, :type => 'timestamp without time zone'
+  else
+    add_index :name, :unique => true
+  end
+end
+if ENV['DB'] == 'postgresql'
+  begin
+    Pet.connection.execute("ALTER TABLE pets DROP CONSTRAINT IF EXISTS unique_name")
+  rescue => e
+    puts e.inspect
   end
-  add_index :name, :unique => true
 end
+
 Pet.auto_upgrade!
 
+if ENV['DB'] == 'postgresql'
```

Comment:
or: @pnomolos you could just get rid of the `disable_native` option and force it to be used when it's possible

that would mean you only have to test with and without `UNIQ_INDEX` (tho you should still do it with env vars instead of duplicating test code)